### PR TITLE
Feat/generator - page generator implemented

### PR DESCRIPTION
**Pull request by pras9**

### CODE REVIEW

File: templates/base/src/routes/index.js [hunk 1]
```diff
@@ -7,3 +7,6 @@ export const NextRouterWrapper = ({ pathName, as, options = {} }) => {
 };
 
 export const getBasePath = (route) => `${BASE_PATH}${route}`;
+
+export const HOMEPAGE = '/';
```
Comment (vinodloha) [inline templates/base/src/routes/index.js:11]: @pras9 These should be maintained in a separate file, I think there is a routeConfig.js

**Comment by vinodloha** [comment]
@pras9 there is an error becasue of missing hooks dir, we should handle such errors. I have added the hooks dir with a .gitkeep for now. But such script breaks should not happen, it should create the dir if it doesn't exist. 

image

**Comment by vinodloha** [comment]
@pras9 for hooks the testing library should be https://www.npmjs.com/package/@testing-library/react-hooks

Hooks can't be tested with React-Testing-Library that you are currently using.

**Comment by vinodloha** [comment]
The page generation is not working as expected. Please make following changes

1. We should ask the page name question first. 
2. Give the url base of the page: (browse) --> Should be optional feature. If no value is passed we should create the page in root page directory. 
3. Give the url base of the page: (browse) --> Where this page should be created, pages or custom dir? (pages)
4. Routes should be added in routes/paths.js

File: templates/base/src/routes/index.js (hunk 1 above)
Comment (pras9) [reply templates/base/src/routes/index.js]: @vinodloha PR comments resolved, PR updated

**Pull request merged**